File: frontend/src/types/rezervasyon.ts
export enum RezervasyonDurum {
  Beklemede  = 0,
  Onaylandi  = 1,
  Reddedildi = 2,
  Iptal      = 3,
}

export const rezervasyonDurumLabel: Record<RezervasyonDurum, string> = {
  [RezervasyonDurum.Beklemede]:  'Beklemede',
  [RezervasyonDurum.Onaylandi]:  'Onaylandı',
  [RezervasyonDurum.Reddedildi]: 'Reddedildi',
  [RezervasyonDurum.Iptal]:      'İptal',
}

export const rezervasyonDurumVariant: Record<RezervasyonDurum, 'default' | 'secondary' | 'destructive' | 'outline'> = {
  [RezervasyonDurum.Beklemede]:  'secondary',
  [RezervasyonDurum.Onaylandi]:  'default',
  [RezervasyonDurum.Reddedildi]: 'destructive',
  [RezervasyonDurum.Iptal]:      'outline',
}

export interface Rezervasyon {
  id: string
  siteId: string
  tesisId?: string | null
  tesisAdi?: string | null
  personId?: string | null
  startDate: string
  endDate: string
  durum: RezervasyonDurum
  notes?: string | null
  createdAt: string
  updatedAt?: string | null
}

export interface CreateRezervasyonDto {
  tesisId?: string | null
  personId?: string | null
  startDate: string
  endDate: string
  durum: RezervasyonDurum
  notes?: string | null
}

export type UpdateRezervasyonDto = CreateRezervasyonDto

File: backend/src/SiteYonetimi.Infrastructure/Entities/Shared/Rezervasyon.cs
namespace SiteYonetimi.Infrastructure.Entities.Shared;

public class Rezervasyon
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid SiteId { get; set; }
    public Guid? TesisId { get; set; }
    public Guid? PersonId { get; set; }
    public DateOnly StartDate { get; set; }
    public DateOnly EndDate { get; set; }
    public RezervasyonDurum Durum { get; set; } = RezervasyonDurum.Beklemede;
    public string? Notes { get; set; }
    public bool IsDeleted { get; set; } = false;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }

    // Nav props
    public Tesis? Tesis { get; set; }
}

public enum RezervasyonDurum
{
    Beklemede = 0,
    Onaylandi = 1,
    Reddedildi = 2,
    Iptal = 3,
}
